teps/0152-tekton-conformance-verification.md

```diff
@@ -0,0 +1,166 @@
+---
+status: implementable
+title: Tekton Conformance Verification
+creation-date: '2024-02-12'
+last-updated: '2024-02-12'
+authors:
+- '@JeromeJu'
+---
+
+# TEP-0152: Tekton Conformance Verification
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Verification Testing Strategy](#verification-testing-strategy)
+    - [Alternatives Considered](#alternatives-considered)
+  - [Vendor Requirements](#vendor-requirements)
+  - [Conformance verification Test coverage](#conformance-verification-test-coverage)
+  - [Test Sutie Design Details](#test-sutie-design-details)
+  - [Tekton Conformance Verification Versioned](#tekton-conformance-verification-versioned)
+- [Future Work](#future-work)
+
+## Summary
+
+This TEP outlines the Tekton Conformance verification process according to [Tekton v1 conformance API spec](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md). It details the implementation of the Tekton conformance verification test suite, paving the way for a community-driven Tekton conformance certification process.
+
+## Motivation
+
+With the release of [v1 Tekton conformance API spec](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md), the Tekton community can now offer Tekton conformance verification. The conformance test suite will be the crucial prerequisite for vendors to claim conformance for their service.
```

Comment:
Did we release a v1 of it ? 🙃

Reply:
That's a great question. As I intended to cover in the [versioning](https://github.com/JeromeJu/community/blob/tep0131-conformance-test/teps/0152-tekton-conformance-verification.md#tekton-conformance-verification-versioned) section, each Conformance policy version should come along with the Pipeline API release version.

At this stage, we haven't officially released V1 yet. But it seems that the goal of [TEP0131](https://github.com/tektoncd/community/blob/main/teps/0131-tekton-conformance-policy.md#proposal) is to initiate Conformance API spec based on existing V1 API. And now [Conformance API spec](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md) is available and [it does targets v1](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md#detailed-resource-types---v1).

It seems to be now coming down to the question do we want to only start offering Conformance verification after the official V1 Pipeline release? (meaning V1 release is a blocker for Conformance?)